uber/api.py

```diff
@@ -0,0 +1,30 @@
+from uber.common import *
+
+attendee_fields = [
+    'full_name', 'first_name', 'last_name', 'email', 'zip_code', 'cellphone', 'ec_phone', 'badge_status_label', 'checked_in',
+    'badge_type_label', 'ribbon_label', 'staffing', 'is_dept_head', 'assigned_depts_labels', 'weighted_hours', 'worked_hours'
+]
+fields = dict({
+    'food_restrictions': {
+        'sandwich_pref_labels': True,
+        'standard_labels': True,
+        'freeform': True
+    },
+    'shifts': {
+        'worked_label': True,
+        'job': ['type_label', 'location_label', 'name', 'description', 'start_time', 'end_time', 'extra15']
+    }
+}, **{field: True for field in attendee_fields})
+
+
+class AttendeeLookup:
+    def lookup(self, badge_num):
+        with Session() as session:
+            attendee = session.query(Attendee).filter_by(badge_num=badge_num).first()
+            return attendee.to_dict(fields) if attendee else {'error': 'No attendee found with Badge #{}'.format(badge_num)}
+
+    def search(self, query):
+        with Session() as session:
+            return [a.to_dict(fields) for a in session.search(query).all()]
+
+services.register(AttendeeLookup(), 'attendee')
```

Comment:
See the Sideboard tutorial for an explanation of how services work:
https://github.com/magfest/sideboard/blob/master/docs/source/index.rst#service

The tl;dr version is that there's a ``sideboard.lib.services`` object, and you can register objects such as modules or class instances, and that will cause Sideboard to expose those as JSON-RPC and WebSocket RPC methods.